packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -976,22 +980,28 @@ export class KeyboardNavigationController extends modules.ViewController {
   }
 
   _enterKeyHandler(eventArgs, isEditing) {
-    const $cell = this._getFocusedCell();
     const rowIndex = this.getVisibleRowIndex();
-    const $row = this._focusedView && this._focusedView.getRow(rowIndex);
+    const key = this._dataController.getKeyByRowIndex(rowIndex);
 
-    if (
-      (this.option('grouping.allowCollapsing') && isGroupRow($row))
-      || (this.option('masterDetail.enabled')
-        && $cell
-        && $cell.hasClass(COMMAND_EXPAND_CLASS))
-    ) {
-      const key = this._dataController.getKeyByRowIndex(rowIndex);
+    const $row = this._focusedView?.getRow(rowIndex);
+    const $cell = this._getFocusedCell();
+
+    const needExpandGroupRow = this.option('grouping.allowCollapsing') && isGroupRow($row);
+    const needExpandMasterDetailRow = this.option('masterDetail.enabled') && $cell?.hasClass(COMMAND_EXPAND_CLASS);
+    const needExpandAdaptiveRow = $cell?.hasClass(ADAPTIVE_COLUMN_NAME_CLASS);
+
+    if (needExpandGroupRow || needExpandMasterDetailRow) {
       const item = this._dataController.items()[rowIndex];
 
-      if (key !== undefined && item && item.data && !item.data.isContinuation) {
+      const isNotContinuation = item?.data && !item.data.isContinuation;
+
+      if (isDefined(key) && isNotContinuation) {
         (this._dataController as any).changeRowExpand(key);
       }
+    } else if (needExpandAdaptiveRow) {
+      this._adaptiveColumnsController.toggleExpandAdaptiveDetailRow(key);
+
+      this._updateFocusedCellPosition($cell);
```

Comment:
Need to update focused cell position, so after toggling next adaptive row focus displayed in correct position